### PR TITLE
add sorting functionality for channels [wip]

### PR DESCRIPTION
**Pull request by spacelatte**
Hello,

I added channel-sorting functionality upon header-click.

I am not really obj-c/mac developer. So feedback is appreciated :)

Here is functionality in-action.

### CODE REVIEW

File: Sources/App/Classes/Dialogs/TDCServerPropertiesSheet.m [hunk 1]
```diff
@@ -2305,6 +2305,15 @@ - (BOOL)tableView:(NSTableView *)tableView acceptDrop:(id <NSDraggingInfo>)info
 	return YES;
 }
 
+- (void)tableView:(NSTableView *)tableView mouseDownInHeaderOfTableColumn:(NSTableColumn *)tableColumn {
+	NSSortDescriptor *sortDescriptor = [[NSSortDescriptor alloc] initWithKey:nil ascending:YES comparator:^NSComparisonResult(IRCChannelConfig *cfg1, IRCChannelConfig *cfg2) {
```
Comment (pvtmert) [inline Sources/App/Classes/Dialogs/TDCServerPropertiesSheet.m]: i dont know if i have to `dealloc` at the end of the function...

Comment (emsquared) [reply Sources/App/Classes/Dialogs/TDCServerPropertiesSheet.m]: Textual uses [Automatic Reference Counting](https://en.wikipedia.org/wiki/Automatic_Reference_Counting) which is fancy speak for it not requiring `dealloc`. The compiler adds that for you.

**Comment by emsquared** [comment]
Thank you for the pull request!

I will review the changes on my next day off.

**Comment by pvtmert** [comment]
Thank you, i have discovered an issue yesterday.

Regarding `[self.channelListArrayController setSortDescriptors];`

If set, user cannot sort channels. (eg needs to save and re-open sheet)
If not set, sorting gets reverted.

I tried to re-assign `self.channelList` but its readonly property

**Pull request closed without merging**